### ISSUE TITLE
Multiple helpers in autowrap

### ISSUE THREAD

**Issue by nathanielvirgo** [sympy/sympy#15337]
The `autowrap` function only supports a single helper function. However, from the documentation and source code, it is clearly designed to support a list of helper functions as well.

The problem seems to be the line

    helpers = [helpers] if helpers else ()

which was added in response to issue #10274. This makes it possible to supply a single helper without putting it inside the list, but if the user tries to supply a list of helpers it puts it inside another list, breaking the code that follows it. It would be super helpful if this can be amended to not put an extra list around the argument if it is a list already.

**Comment by Bavish2201** [comment]
Hello, I would like to work on this issue. As I'm a beginner, would you please help me with this?
According to what I understood, I should change the above mentioned code to 
```
if type(helpers) is not list: 
    helpers = [helpers] if helpers else ()
```
Did I miss anything?

**Comment by moorepants** [comment]
@Bavish2201 You have the right idea. Make sure to write a unit test for the various possible helper inputs.

**Comment by nathanielvirgo** [comment]
I am not an expert either, but I would rather say `if type(helpers) is not tuple`, since someone might want to pass in some other iterable instead of a list.

**Comment by Bavish2201** [comment]
Would this be better: `if not hasattr(helpers, '__iter__')`

**Comment by moorepants** [comment]
@Bavish2201 I recommend writing the improved unit tests for autowrap first, then work on the solution.

**Comment by asmeurer** [comment]
A better check would be `if callable(helpers): helpers = [helpers]`.

**Comment by nathanielvirgo** [comment]
@asmeurer that won't work - the helpers are supplied as tuples, not callables. So I guess it has to wrap it in a list if the argument is a tuple, but not if it's any other kind of iterable. (Or possibly it would be better to just change it back to using iterables of tuples and change the unit tests to reflect that.)

**Comment by FrackeR011** [comment]
@Bavish2201 are you still working on this

**Comment by asmeurer** [comment]
There is an `iterable` function in the sympy.core.compatibility module you can use to check if something is iterable.

**Comment by Bavish2201** [comment]
@asmeurer  When `helpers` is just one `tuple`, we need to do `helpers = [helpers]`, but if `helpers` is a list of tuples, then we shouldn't do this. But `iterable(helpers)` will be `true` in both the cases. How to identify when `helpers` is just one tuple or a tuple/list of tuples? Please help.

**Comment by asmeurer** [comment]
I see. I don't see why the original change was needed in the first place.

**Comment by nathanielvirgo** [comment]
I think the original change was made because the documentation was unclear. The OP of #10274 was trying to call autowrap with `helpers=('helper',x-y+z, (x,z,y))` when they should have been doing `helpers=[('helper',x-y+z, (x,z,y))]`, which I am guessing would have worked.

In response (changes #10274) a unit test was added to support the `helpers=('helper',x-y+z, (x,z,y))` syntax and the code was changed to pass this test. But I think that unit test is actually wrong, and it should be `helpers=[('helper',x-y+z, (x,z,y))]` instead. Also there should be another test with multiple helpers, e.g. `helpers=[('helper1',x-y+z, (x,z,y)), ('helper2',sin(u)+1,(u,))]`.

I suggested changing things to allow both syntaxes because people might have code that relies on the current behaviour. But if that's not a concern I would definitely vote for reverting the change and making it so you always have to write `helpers=[('helper',x-y+z, (x,z,y))]` (as well as clearly documenting this!). That just makes more sense, and it makes it possible to supply a tuple of helper functions instead of a list.

asmeurer, could you make a call on that?

**Comment by asmeurer** [comment]
I agree the original change was not necessary. I'm not sure if we should just change back or try to support both. 

CC @tokencolour @megh1241 would changing this back break your code?

**Comment by moorepants** [comment]
I've submitted a fix in #15391.
